src/celeritas/phys/PhysicsParams.cc

```diff
@@ -231,14 +233,16 @@ void PhysicsParams::build_ids(const ParticleParams& particles,
 {
     CELER_EXPECT(data);
     CELER_EXPECT(!models_.empty());
-    using ModelRange = std::tuple<real_type, real_type, ModelId>;
+    using ModelRange = std::tuple<real_type, real_type, ParticleModelId>;
+
+    // Offset from the index in the list of models to a model's ActionId
+    data->scalars.model_to_action = this->model(ModelId{0}).action_id().get();
 
     // Note: use map to keep ProcessId sorted
     std::vector<std::map<ProcessId, std::vector<ModelRange>>> particle_models(
         particles.size());
-
-    // Offset from the index in the list of models to a model's ActionId
-    data->scalars.model_to_action = this->model(ModelId{0}).action_id().get();
+    std::vector<ModelId> temp_model_ids;
+    size_type            pm_idx{0};
```

Comment:
Even though they're (currently) the same, I try to specifically use `FooId::size_type` because that guarantees you're not narrowing when converting with `FooId{counter}`, as would be the case if we decided to change `ParticleModelId` to a short uint or something.
```suggestion
    ParticleModelId::size_type            pm_idx{0};
```